src/NuGet.Licenses/Services/LicenseFileService.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Web;
+namespace NuGet.Licenses
+{
+    public class LicenseFileService : ILicenseFileService
+    {
+        public string GetLicenseFilePath(string licenseIdentifier)
+        {
+            return Path.GetFullPath(Path.Combine(HttpContext.Current.Server.MapPath("~\\App_Data\\Licenses\\"), String.Concat(licenseIdentifier, ".txt")));
+        }
+
+        public bool IsLicenseFilePathAllowed(string licenseFilePath)
```

Comment:
I'd suggest to remove this method and instead always check for path correctness in `GetLicenseFilePath` and throw an exception if it's not correct. Then calling code would not have a chance not to check for the path correctness.

Reply:
Done